lightkurve/tests/test_synthetic_data.py

```diff
@@ -0,0 +1,188 @@
+    const, sin_weight, cos_weight = least_squares_coeffs
+
+    fractional_amplitude = (sin_weight**2+cos_weight**2)**(0.5) / const
+    assert ((fractional_amplitude > true_amplitude/2) &
+            (fractional_amplitude < true_amplitude*2) )
+
+@pytest.mark.skipif(lacks_BLS, reason="Astropy BLS requires Python 3")
+def test_transit_SFF():
+    """Test the SFF implementation with known signals"""
+    # Retrieve the custom, known signal properties
+    tpf = KeplerTargetPixelFile(filename_synthetic_transit)
+    true_period = np.float(tpf.hdu[3].header['PERIOD'])
+    true_depth = np.float(tpf.hdu[3].header['RPRS'])**2.0
+
+    # Run the SFF algorithm
+    lc = tpf.to_lightcurve()
+    corrector = SFFCorrector(lc)
+    cor_lc = corrector.correct(tpf.pos_corr2, tpf.pos_corr1,
+                                 niters=4, windows=5, bins=7, restore_trend=False)
+
+    # Verify that we get the transit period within 5%
+    pg = cor_lc.to_periodogram(method='bls', minimum_period=1, maximum_period=9,
+                               frequency_factor=0.05, duration=np.arange(0.1, 0.6, 0.1))
+    ret_period = pg.period_at_max_power.value
+    threshold = 0.05
+    assert ((ret_period > true_period*(1-threshold)) &
+            (ret_period < true_period*(1+threshold)) )
+
+    # Verify that we get the transit depth to within a factor of 50%
+    assert ((pg.depth_at_max_power > true_depth/1.5) &
+            (pg.depth_at_max_power < true_depth*1.5))
+
+
+@pytest.mark.skipif(lacks_BLS, reason="Astropy BLS requires Python 3")
+def test_transit_PLD():
+    """Test the SFF implementation with known signals"""
+    # Retrieve the custom, known signal properties
+    tpf = KeplerTargetPixelFile(filename_synthetic_transit)
+    true_period = np.float(tpf.hdu[3].header['PERIOD'])
+    true_depth = np.float(tpf.hdu[3].header['RPRS'])**2.0
+
+    # Run the PLD algorithm on a first pass
+    corrector = PLDCorrector(tpf)
+    cor_lc = corrector.correct(use_gp=False)
+    pg = cor_lc.to_periodogram(method='bls', minimum_period=1, maximum_period=9,
+                               frequency_factor=0.05, duration=np.arange(0.1, 0.6, 0.1))
+
+    # Re-do PLD with the suspected transits masked
+    cor_lc = corrector.correct(use_gp=False, cadence_mask=pg.get_transit_mask()).normalize()
+    pg = cor_lc.to_periodogram(method='bls', minimum_period=1, maximum_period=9,
+                               frequency_factor=0.05, duration=np.arange(0.1, 0.6, 0.1))
+
+    # Verify that we get the period within ~5%
+    ret_period = pg.period_at_max_power.value
+    threshold = 0.05
+    assert ((ret_period > true_period*(1-threshold)) &
+            (ret_period < true_period*(1+threshold)) )
+
+    # Verify that we get the transit depth to within a factor of 50%
+
+    print(pg.depth_at_max_power, true_depth)
+    assert ((pg.depth_at_max_power > true_depth/1.5) &
+            (pg.depth_at_max_power < true_depth*1.5))
```

Comment:
I see PLD only recovers the depth to ~15% or so on my machine, so let's make this test more stringent?  Ideally, can you post a quick plot of ideal vs detrended light curve to make sure nothing crazy is going on?